app_integrations/apps/aliyun.py

```diff
@@ -0,0 +1,148 @@
+            request.set_NextToken(self._next_token)
+
+        try:
+            response = client.do_action_with_exception(request)
+            json_response = json.loads(response)
+            if 'NextToken' in json_response:
+                self._more_to_poll = True
+                self._next_token = json_response['NextToken']
+            else:
+                self._more_to_poll = False
+                self._last_timestamp = json_response['EndTime']
+
+            return json_response['Events']
+
+        except ServerException as e:
+            LOGGER.exception("Server Exception: %s", str(e))
+            return False
+        except ClientException as e:
+            LOGGER.exception("Client Exception: %s", str(e))
+            return False
+
+    @classmethod
+    def _required_auth_info(cls):
+        """Required credentials for access to the resources"""
+
+        def region_validator(region):
+            """Region names pulled from https://www.alibabacloud.com/help/doc-detail/40654.htm"""
+
+            if region in ["cn-qingdao", "cn-beijing", "cn-zhangjiakou", "cn-huhehaote",
+                          "cn-hangzhou", "cn-shanghai", "cn-shenzhen", "cn-hongkong",
+                          "ap-southeast-1", "ap-southeast-2", "ap-southeast-3", "ap-southeast-5",
+                          "ap-northeast-1", "ap-south-1", "us-west-1", "us-east-1",
+                          "eu-central-1", "me-east-1"]:
+                return region
+            return False
+
+        return {
+            'access_key_id': {
+                'description': ('The access key id generated for a RAM user. This '
+                                'should be a string of alphanumeric characters.'),
+                'format': re.compile(r'.*')
+                },
+            'access_key_secret': {
+                'description': ('The access key secret generated for a RAM user. This '
+                                'should be a string of alphanumeric characters.'),
+                'format': re.compile(r'.*')
+                },
+            'region_id': {
+                'description': ('The region for the Aliyun API. This should be '
+                                'a string like \'ap-northeast-1\'.'),
+                'format': region_validator
+                },
+            }
+
+    @classmethod
+    def _sleep_seconds(cls):
+        """Return the number of seconds this polling function should sleep for
+        between requests to avoid failed requests. The Aliyun documentation doesn't
+        list limits on the requests portion of the actionTrail feature, so the only
+        limit is the general limit on Aliyun API requests, which is no more than
+        100 per second. We can set this value to 0 safely.
+
+        Resource:
+            https://www.alibabacloud.com/help/doc-detail/29474.htm
+
+        Returns:
+            int: Number of seconds the polling function should sleep for
+        """
+        return 0
```

Comment:
I think if this is 0 it can be left off completely. I do like the context you've added in the docstring, though, so I'm not sure if I feel strongly about removing the method